tests/test_update.py

```diff
@@ -261,6 +269,47 @@ def test_effective_user(self, update):
         else:
             assert user is None
 
+    def test_effective_sender(self, update):
+        # Test that it's sometimes None per docstring
+        sender = update.effective_sender
+        if not (
+            update.channel_post is not None
+            or update.edited_channel_post is not None
+            or update.poll is not None
+            or update.chat_boost is not None
+            or update.removed_chat_boost is not None
+            or update.message_reaction_count is not None
+        ):
+            assert sender.id == 1
+        else:
+            assert sender is None
+
+    def test_effective_sender_anonymous(self, update):
+        # remove none anonymous user
+        if message := (update.message or update.edited_message):
+            message._unfreeze()
+            message.from_user = None
+        elif reaction := (update.message_reaction):
+            reaction._unfreeze()
+            reaction.user = None
+        elif answer := (update.poll_answer):
+            answer._unfreeze()
+            answer.user = None
```

Comment:
TBH I don't quite get what this test is covering that the above tests does not cover 🤔

Reply:
ahhh, here i was trying to mimic updates from anonymous group members. for example a reaction coming from  anonymous user would have `actor_chat` and not `user`. but since they're both set, i remove `user`.
this may allow us to do `assert isinstance(sender, Chat)` you mentioned